articles/configure-custom-routing.mdx

```diff
@@ -0,0 +1,209 @@
+<comp.Notice info_type="info">
+  CloudCannon reads your routes in order, and will use the earliest matching rule for a given `from` URL. Duplicate rules are ignored.
+</comp.Notice>
+
+<comp.Notice info_type="info">
+  You can redirect to, but not from a URL fragment (e.g. example.com/blog/#heading). The URL fragment is not passed to the server so, it cannot be processed by the redirect.
+</comp.Notice>
+
+<comp.Notice info_type="info">
+  You can use routing to load a single HTML file on multiple URL routes and use JavaScript to load different content.
+</comp.Notice>
+
+You must configure three keys for a custom redirect: the `from` URL, the `to` URL, and the `status` code. Proxies do not require the `to` key.
+
+* **From** — Specifies which URL should trigger this redirect rule.
+* **To** — Specifies which URL the visitor should be redirected to.
+* **Status** — Tells the visitor's browser the reason for the route using an [HTTP status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes).
+
+To add custom routes to your `routing.json` file:
+
+1. Navigate to the `.cloudcannon` folder in your *File browser* and click on the `routing.json` file. CloudCannon will open `routing.json` in the Source Editor.
+2. &nbsp;Create a JSON array under the `routes` key. Each entry in the array could include the keys `from`, `to`, and `status`.
+3. When you are happy with your routing file, save your changes.
+
+After a successful build, CloudCannon will route visitors according to the rules in your `routing.json` file.
+
+Here are a few examples:
+
+<comp.CodeBlock language="json" source=".cloudcannon/routing.json">
+```
+
+{
+  "routes": [
+    {
+      "from": "/cloudcannon/(.*)",
+      "to": "https://cloudcannon.com/$1",
+      "status": 302
+    },
+    {
+      "from": "/documentation/articles/choosing-your-ssg/",
+      "to": "/documentation/articles/select-your-ssg/",
+      "status": 301
+    },
+    {
+      "from": "/app/(.*)",
+      "to": "/app/",
+      "status": 200
+    }
+  ]
+}
+
+```
+</comp.CodeBlock>
+
+For more a complete list of configuration keys available for custom routing, please read our [routing reference documentation](/documentation/articles/routing-reference/).
+
+### Generating redirects
+
+If your Site requires it, you can generate your routing file at build time.
+
+For example, if you only want to redirect URLs on your production Site, you could add the following Liquid templating:
+
+<comp.CodeBlock language="JSON" source="routing.json">
+``````````
+
+{
+  routes: [
+    {% if environment == 'production' %}
+      {from: '/example-1',
+      to: '/example-2'}
+    {% endif %}
+  ]
+}
+
+``````````
+</comp.CodeBlock>
+
+You can create a build process to generate your routing file with the correct content, depending on the environment. In this case, the build process should generate the file in  `_cloudcannon/routing.json` . CloudCannon will use this file instead of  `.cloudcannon/routing.json` from your source.
```

Comment:
```suggestion
You can create a build process to generate your routing file with the correct content, depending on the environment. In this case, the build process should generate the file to `_cloudcannon/routing.json` in your site output directory. CloudCannon will use this file instead of  `.cloudcannon/routing.json` from your source.
```